File: app/src/main/java/com/example/jvmori/myweatherapp/di/modules/ActivityBuildersModule.java
package com.example.jvmori.myweatherapp.di.modules;

import com.example.jvmori.myweatherapp.MainActivity;
import com.example.jvmori.myweatherapp.di.modules.main.MainFragmentBuildersModule;
import com.example.jvmori.myweatherapp.di.modules.main.WeatherViewModelsModule;
import com.example.jvmori.myweatherapp.di.scope.MainActivityScope;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class ActivityBuildersModule {
    @ContributesAndroidInjector(
            modules = {
                    WeatherViewModelsModule.class,
                    MainFragmentBuildersModule.class
            }
    )
    abstract MainActivity contributeMainActivity();
}

File: app/src/main/java/com/example/jvmori/myweatherapp/di_/modules/ActivityBuildersModule.kt
package com.example.jvmori.myweatherapp.di_.modules

import com.example.jvmori.myweatherapp.MainActivity
import com.example.jvmori.myweatherapp.di_.modules.main.WeatherViewModelsModule
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class ActivityBuildersModule {

    @ContributesAndroidInjector(
            modules = [WeatherViewModelsModule::class]
    )
    abstract fun contributeMainActivity(): MainActivity
}